lib/spack/spack/util/environment.py

```diff
@@ -56,7 +56,7 @@
     "csh": "unsetenv {0};\n",
     "fish": "set -e {0};\n",
     "bat": 'set "{0}="\n',
-    "pwsh": "Remove-Item Env:{0}\n",
+    "pwsh": "Set-Item -Path Env:{0}\n",
```

Comment:
Is this the same as `Remove-Item` or does it avoid some error?

Apologies if I managed to learn nothing from https://github.com/spack/spack/pull/39118#discussion_r1304751174

Reply:
IIRC, it was due to a small corner case where we're unsetting variables that don't exist. In that case, `Set-Item` is a no-op and `Remove-Item` fails and crashes the env operation.